src/modules/rover_ackermann/RoverAckermann.cpp

```diff
@@ -53,174 +53,113 @@ bool RoverAckermann::init()
-
-				if (fabsf(rover_ackermann_setpoint.lateral_acceleration_setpoint) > FLT_EPSILON
-				    || fabsf(rover_ackermann_setpoint.forward_speed_setpoint) < FLT_EPSILON) { // Closed loop yaw rate control
-					_course_control = false;
-
-				} else { // Course control if the steering input is zero (keep driving on a straight line)
-					if (!_course_control) {
-						_pos_ctl_course_direction = Vector2f(cos(_vehicle_yaw), sin(_vehicle_yaw));
-						_pos_ctl_start_position_ned = _curr_pos_ned;
-						_course_control = true;
-					}
-
-					// Construct a 'target waypoint' for course control s.t. it is never within the maximum lookahead of the rover
-					const float vector_scaling = sqrtf(powf(_param_pp_lookahd_max.get(),
-										2) + powf(_posctl_pure_pursuit.getCrosstrackError(), 2)) + _posctl_pure_pursuit.getDistanceOnLineSegment();
-					const Vector2f target_waypoint_ned = _pos_ctl_start_position_ned + sign(
-							rover_ackermann_setpoint.forward_speed_setpoint) *
-									     vector_scaling * _pos_ctl_course_direction;
-					// Calculate steering setpoint
-					const float steering_setpoint = _ackermann_guidance.calcDesiredSteering(_posctl_pure_pursuit,
-									target_waypoint_ned, _pos_ctl_start_position_ned, _curr_pos_ned, _param_ra_wheel_base.get(),
-									rover_ackermann_setpoint.forward_speed_setpoint, _vehicle_yaw, _param_ra_max_steer_angle.get(), _armed);
-					rover_ackermann_setpoint.lateral_acceleration_setpoint = powf(_vehicle_forward_speed,
-							2.f) * tanf(steering_setpoint) / _param_ra_wheel_base.get();
-				}
-
-				_rover_ackermann_setpoint_pub.publish(rover_ackermann_setpoint);
-			}
-
-		} break;
-
-	case vehicle_status_s::NAVIGATION_STATE_AUTO_MISSION:
-	case vehicle_status_s::NAVIGATION_STATE_AUTO_RTL:
-		_ackermann_guidance.computeGuidance(_vehicle_forward_speed, _vehicle_yaw, _nav_state, _armed);
-		break;
-
-	default: // Unimplemented nav states will stop the rover
-		rover_ackermann_setpoint_s rover_ackermann_setpoint{};
-		rover_ackermann_setpoint.timestamp =  timestamp;
-		rover_ackermann_setpoint.forward_speed_setpoint =  NAN;
-		rover_ackermann_setpoint.forward_speed_setpoint_normalized =  0.f;
-		rover_ackermann_setpoint.steering_setpoint = NAN;
-		rover_ackermann_setpoint.steering_setpoint_normalized = 0.f;
-		rover_ackermann_setpoint.lateral_acceleration_setpoint = NAN;
-		_rover_ackermann_setpoint_pub.publish(rover_ackermann_setpoint);
-		break;
+	hrt_abstime timestamp_prev = _timestamp;
+	_timestamp = hrt_absolute_time();
+	_dt = math::constrain(_timestamp - timestamp_prev, 1_ms, 5000_ms) * 1e-6f;
+
+	_ackermann_pos_control.updatePosControl();
+	_ackermann_att_control.updateAttControl();
+	_ackermann_rate_control.updateRateControl();
+
+	if (_vehicle_control_mode_sub.updated()) {
+		_vehicle_control_mode_sub.copy(&_vehicle_control_mode);
 	}
 
-	if (!_armed) {
-		_ackermann_control.resetControllers();
+	bool manual_mode_enabled = _vehicle_control_mode.flag_control_manual_enabled
```

Comment:
```suggestion
	bool full_manual_mode_enabled = _vehicle_control_mode.flag_control_manual_enabled
```
To distinguish it from the other "manual" modes that still contain some controller support.

Reply:
Suggestion applied in [454995a](https://github.com/PX4/PX4-Autopilot/pull/24285/commits/454995a3a15f362273c6bc3da6fc9edb9e088165)